lib/geo.js

```diff
@@ -10,23 +10,31 @@ var assert = require('assert');
  */
 
 exports.nearFilter = function nearFilter(where) {
+  var keyPath = [];
   var result = false;
 
-  if (where && typeof where === 'object') {
-    Object.keys(where).forEach(function(key) {
-      var ex = where[key];
-
-      if (ex && ex.near) {
-        result = {
-          near: ex.near,
-          maxDistance: ex.maxDistance,
-          unit: ex.unit,
-          key: key,
-        };
-      }
-    });
+  searchForNear(where);
+
+  function searchForNear(node) {
+    if (node && typeof node === 'object') {
+      Object.keys(node).forEach(function(key) {
+        var ex = node[key];
+        keyPath.push(key);
+
+        if (ex && ex.near) {
+          result = {
+            near: ex.near,
+            maxDistance: ex.maxDistance,
+            unit: ex.unit,
+            key: keyPath.slice(0),
```

Comment:
I'm not very familiar with `key` I could not find any documentation about this; what is supposed to return? @0candy @superkhau @jannyHou